src/rulesets/rest/2022-05-25/property-rules.ts

```diff
@@ -209,9 +209,14 @@ const arrayWithItemsInRequest = new RequestRule({
       (property) => {
         if (property.raw.type === "array") {
           if (!property.raw.items || !("type" in property.raw.items)) {
-            throw new RuleError({
-              message: "type was not found array items",
-            });
+            const oneOf = property.raw.items["oneOf"];
+            const allOf = property.raw.items["allOf"];
+            const anyOf = property.raw.items["anyOf"];
+            if (!oneOf && !allOf && !anyOf) {
+              throw new RuleError({
+                message: "type was not found array items",
+              });
+            }
```

Comment:
Recommend turning this into a helper, for incremental improvement as described above. Asserting a non-empty type seems like a generally useful thing we may want for property types as well as array item types.